### PR TITLE
Updating the broken scaleway cloud module

### PR DESCRIPTION
**Pull request by rjeschmi**
### Fixes some issues with the scaleway module

Looking for feedback to get this module up to standard.

### Previous Behavior
scaleway module didn't work

### New Behavior
Works fine for listing images and servers

### Tests written?

No

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by cachedout** [comment]
Hi @rjeschmi. Could you please fix these style issues? https://jenkinsci.saltstack.com/job/pr-lint/job/PR-50334/3/warnings52Result/new/

File: salt/cloud/clouds/scaleway.py [hunk 1]
```diff
@@ -313,16 +314,22 @@ def __query_node_data(server_name):
     return ret
 
 
-def query(method='servers', server_id=None, command=None, args=None,
-          http_method='get'):
+def query(method='servers', root='api_root', server_id=None, command=None, args=None,
```
Comment (gtmanfred) [inline salt/cloud/clouds/scaleway.py]: We do not break stuff where people use positional args, so this `root` needs to be moved to the end of the function definition.

File: salt/cloud/clouds/scaleway.py [hunk 2]
```diff
@@ -78,8 +78,9 @@ def avail_images(call=None):
             '-f or --function, or with the --list-images option'
         )
 
-    items = query(method='images')
+    items = query(method='images', root='marketplace_root')
     ret = {}
+    log.debug("items type: %s", type(items))
```
Comment (gtmanfred) [inline salt/cloud/clouds/scaleway.py]: This seems like this was used while making these changes? does it still need to be here, it doesn't seem useful.

File: salt/cloud/clouds/scaleway.py [hunk 3]
```diff
@@ -345,25 +352,30 @@ def query(method='servers', server_id=None, command=None, args=None,
...
+                                      decode=True,
+                                      decode_type='json',
+                                      data_render=True,
+                                      data_renderer='json',
+                                      headers=True,
+                                      header_dict={'X-Auth-Token': token,
                                                'User-Agent': "salt-cloud",
                                                'Content-Type': 'application/json'})
-    if request.status_code > 299:
+    if request['status'] > 299:
```
Comment (gtmanfred) [inline salt/cloud/clouds/scaleway.py]: What is the purpose for this change?

**Review by gtmanfred — CHANGES_REQUESTED** [review]

File: salt/cloud/clouds/scaleway.py (hunk 1 above)
Comment (rjeschmi) [reply salt/cloud/clouds/scaleway.py]: sure, thanks for the feedback

File: salt/cloud/clouds/scaleway.py (hunk 3 above)
Comment (rjeschmi) [reply salt/cloud/clouds/scaleway.py]: There was a change in the util I think (maybe a long time ago?). This module hasn't worked in a while. checking against request['status'] seems to be pretty standard in the other providers. There is some inconsistency in testing for error, but this is probably close.

File: salt/cloud/clouds/scaleway.py (hunk 2 above)
Comment (rjeschmi) [reply salt/cloud/clouds/scaleway.py]: yep, removed

**Comment by rjeschmi** [comment]
There are still issues with the functionality. I'll test and let you know when I have everything fixed.

File: salt/cloud/clouds/scaleway.py (hunk 3 above)
Comment (gtmanfred) [reply salt/cloud/clouds/scaleway.py]: perfect, thanks, just wanted to make sure.

**Review by gtmanfred — APPROVED** [review]

**Comment by dcharbonnier** [comment]
This should be released it's already a big improvement compared to the current version

**Comment by robertscholts** [comment]
Does this means its already in the current saltstack version?

**Comment by terminalmage** [comment]
Ported to master in https://github.com/saltstack/salt/pull/56709

**Comment by terminalmage** [comment]
@robertscholts Unfortunately not. The project changed the default branch from which releases are cut to `master` and so there are things that were merged into this `develop` branch which did not make it into a Salt release yet.

I've ported this PR to `master` which will get it into a future release, follow the PR I linked above to track its progress.
